scipy/spatial/distance.py

```diff
@@ -1395,6 +1451,10 @@ def dfun(u, v):
 
     elif isinstance(metric, string_types):
         mstr = metric.lower()
+        if w is not None and not mstr.startswith("test_"):
+            if mstr in ['seuclidean', 'se', 's', 'mahalanobis']:
+                raise ValueError("metric %s incompatible with weights" % mstr)
+            mstr = "test_%s" % mstr
```

Comment:
I think that then it could be useful to suggest the user to use the python version instead if he needs weights.